src/bitfield.cpp

```diff
@@ -190,5 +190,118 @@ namespace libtorrent
 		clear_trailing_bits();
 		TORRENT_ASSERT(size() == bits);
 	}
+
+	int bitfield::find_first_set() const
+	{
+		const int words = num_words();
+		if (words == 0) return -1;
+		std::uint32_t const* const ptr = &m_buf[1];
+		for (int i = 0; i < words; ++i)
+		{
+			if (ptr[i] == 0) continue;
+
+			// now, find the index of the first set bit in ptr[i]
+			uint32_t v = aux::network_to_host(ptr[i]);
+
+#if TORRENT_HAS_SSE
```

Comment:
just to leave a note, now there is `TORRENT_HAS_BUILTIN_CLZ` and can be seen in use at `sha1_hash::count_leading_zeroes`